apps/agreement/contracts/Agreement.sol

```diff
@@ -137,9 +140,18 @@ contract Agreement is IAgreement, AragonApp {
     * @param _title String indicating a short description
     * @param _content Link to a human-readable text that describes the initial rules for the Agreements instance
     * @param _arbitrator Address of the IArbitrator that will be used to resolve disputes
+    * @param _aragonAppFeesCashier Transaction fees oracle to fetch fees for new actions
     * @param _stakingFactory Staking factory to be used for the collateral staking pools
     */
-    function initialize(string _title, bytes _content, IArbitrator _arbitrator, StakingFactory _stakingFactory) external {
+    function initialize(
+        string _title,
+        bytes _content,
+        IArbitrator _arbitrator,
+        IAragonAppFeesCashier _aragonAppFeesCashier,
```

Comment:
Just wondering, but is there a way to get this off of the `IArbitrator` interface, so there's less chances of getting it wrong?

Reply:
We have been thinking about it and didn’t find a clean solution, that’s why we went this route. The main problem would be upgrading the `Controller` in Aragon Court.